usr/sbin/p11sak/p11sak.c

```diff
@@ -1878,11 +1868,6 @@ static CK_RV parse_remove_key_args(char *argv[], int argc, p11sak_kt *kt,
         *label = "";
     }
 
-    if (*slot == 0) {
```

Comment:
So now you don't recognize anymore that no slot is specified, but silently use slot 0. This might be OK if this is intended.....

You could alternatively have a flag variable within this function that is set to true when the slot option is parsed, and left false when no slot option is found. Then you could check that flag here and issue the message if no slot was specified. 

The same applies to the other 2 places that you changed.

Reply:
I added a bool to check if the slot argument was set in all three functions that check for arguments.